src/common/io.py

```diff
@@ -1324,17 +1358,20 @@ def _get_name_dict(self):
 
     def _get_trajectory(self, data_index):
         if isinstance(self._data_2, dict):
-            if data_index in self._data_2:
-                return self._data_2[data_index]
+            if self._mtime != os.path.getmtime(self._fname):
+                if self._allow_file_updates:
+                    self._update_data_info()
+                else:
+                    raise JIOError("The result file have been modified since the result object was created. Please make sure that different filenames are used for different simulations.")
```

Comment:
I think we should put the "raise JIOError" in a separate function now because this code is duplicated with line 1321.